### ISSUE TITLE
Download manager

### ISSUE THREAD

**Issue by csziacobus** [eudoxia0/trivial-download#2]
Should spin up a new thread for every download process, enabling more than one download at a time.

**Comment by eudoxia0** [comment]
Hm. Perhaps there should be a function that takes a list of `(url . pathname)` conses and downloads each them in parallel, adjusting that to the number of actual cores.

**Comment by eudoxia0** [comment]
Well having thought about it this is out of scope.
